### PR TITLE
Add webcfg-Plugin, Reload config on USR1-sig, SharedConfig class and more...

### PR DESCRIPTION
**Pull request by dadav**
So, the first idea with the new on_webhook was to create a site in the browser to change the config. The problem was, that the config was all over the place (multiple threads etc.). This is the reason why i created the SharedConfig class. It stores the config in an internal dict, in one place, which allows us to change the config in all threads. What? Change things which are used by multiple threads? Yes, i know. Thats why there is a RWLock on it :)

### CODE REVIEW

**Comment by dadav** [comment]
Also, big THX to Azkrath and RoM who helped with the html/javascript stuff^^

**Comment by dadav** [comment]
related issues: #206

File: pwnagotchi/plugins/__init__.py [hunk 1]
```diff
@@ -44,16 +44,19 @@ def load_from_file(filename):
     return plugin_name, instance
 
 
-def load_from_path(path, enabled=()):
+def load_from_path(path, enabled=list()):
```
Comment (neutralinsomniac) [inline pwnagotchi/plugins/__init__.py]: So this is (probably) not gonna do what you want here. If you call this function with no 'enabled', argument, python will use the *same* (mutable) list object every time you call this function. Check this out:

https://docs.python-guide.org/writing/gotchas/

and my simple example:
test.py:
```
def wtf(arg=list()):
    arg.append('a')
    return arg

print wtf()
print wtf()
```

```
jeremy@x1:~/pwnagotchi$ python test.py
['a']
['a', 'a']
```

Comment (dadav) [reply pwnagotchi/plugins/__init__.py]: very good catch!

**Pull request closed without merging**